### PR TITLE
integrate RWA improvements

### PR DESCRIPTION
**Pull request by bwbohl**
@roewenstrunk @nikobeer : any chance we get back to a sane development concept of Edirom Online, reintegrating improvements from edition projects?

For example RWA has made huge improvements concerning localisation, nevertheless there are some RWA specific things in the code, an some things that all projects seem to change for their purpose e.g. the value of ActiveWork in Application.js

Could we get together to discuss a strategy?

### CODE REVIEW

File: app/controller/window/SingleWindowController.js [hunk 1]
```diff
@@ -71,15 +73,32 @@ Ext.define('EdiromOnline.controller.window.SingleWindowController', {
         var me = this;
         var views = [];
         
+        //console.log("onMetaDataLoaded config");
+        //console.log(config);
+
```
Comment (roewenstrunk) [inline app/controller/window/SingleWindowController.js:78]:
```suggestion
```

File: app/controller/window/SingleWindowController.js [hunk 2]
```diff
@@ -95,6 +114,9 @@ Ext.define('EdiromOnline.controller.window.SingleWindowController', {
         var id = type;
         var label = (config.viewLabel && config.viewLabel != ''?config.viewLabel:me.getLabel(type));
         var viewClass = me.getViewClass(type);
+        
+        //console.log("createView");
+        //console.log(config);
```
Comment (roewenstrunk) [inline app/controller/window/SingleWindowController.js:119]:
```suggestion
```

File: app/controller/window/iFrameView.js [hunk 3]
```diff
@@ -41,6 +41,9 @@ Ext.define('EdiromOnline.controller.window.iFrameView', {
 
         var uri = view.uri;
         
+        //console.log("iFrameView:onAfterLayout");
+        //console.log(view);
+
```
Comment (roewenstrunk) [inline app/controller/window/iFrameView.js:46]:
```suggestion
```

File: app/view/window/AnnotationView.js [hunk 4]
```diff
@@ -60,27 +60,37 @@ Ext.define('EdiromOnline.view.window.AnnotationView', {
                 filters: []
             }],
             columns: [
+// 	            Not needed for RWA
```
Comment (roewenstrunk) [inline app/view/window/AnnotationView.js:63]:
```suggestion
```

File: app/view/window/AnnotationView.js [hunk 5]
```diff
@@ -60,27 +60,37 @@ Ext.define('EdiromOnline.view.window.AnnotationView', {
...
+/*
```
Comment (roewenstrunk) [inline app/view/window/AnnotationView.js:64]:
```suggestion
```

File: app/view/window/AnnotationView.js [hunk 6]
```diff
@@ -60,27 +60,37 @@ Ext.define('EdiromOnline.view.window.AnnotationView', {
...
             	{
                     header: getLangString('view.window.AnnotationView_No'),
                     dataIndex: 'pos',
                     width: 35
                 },
+*/
```
Comment (roewenstrunk) [inline app/view/window/AnnotationView.js:70]:
```suggestion
```

File: app/view/window/AnnotationView.js [hunk 7]
```diff
@@ -432,8 +442,8 @@ Ext.define('EdiromOnline.view.window.AnnotationView', {
         });
 
         me.setPreviewGrid(participants);
-        me.setPreviewSingle(participants);
-        me.setPreviewList(participants);
+/*        me.setPreviewSingle(participants);*/
+/*        me.setPreviewList(participants);*/
```
Comment (roewenstrunk) [inline app/view/window/AnnotationView.js:446]:
```suggestion
        me.setPreviewSingle(participants);
        me.setPreviewList(participants);
```

File: app/view/window/AnnotationView.js [hunk 8]
```diff
@@ -469,24 +479,29 @@ Ext.define('EdiromOnline.view.window.AnnotationView', {
             var page = participant['page'];
             var source = participant['source'];
             var siglum = participant['siglum'];
-            var digilibBaseParams = participant['digilibBaseParams'];
-            var digilibSizeParams = participant['digilibSizeParams'];
+            var part = participant['part'];
+            // var digilibBaseParams = participant['digilibBaseParams'];
+            // var digilibSizeParams = participant['digilibSizeParams'];
+            var digilibURL = participant['digilibURL'];
```
Comment (roewenstrunk) [inline app/view/window/AnnotationView.js:485]:
```suggestion
            var digilibBaseParams = participant['digilibBaseParams'];
            var digilibSizeParams = participant['digilibSizeParams'];
```

File: app/view/window/AnnotationView.js [hunk 9]
```diff
@@ -469,24 +479,29 @@ Ext.define('EdiromOnline.view.window.AnnotationView', {
...
+                   loadLink(participant.linkUri, {});
+                });
+                
                 elems.push(shape);
 
                 square |= true;
 
             }else {
-                var shape = tplImg.append(div, [id, digilibBaseParams + "dw=600&amp;amp;dh=600" + digilibSizeParams, hiddenData, label], true);
+                var shape = tplImg.append(div, [id, /* digilibBaseParams + "dw=600&amp;amp;dh=600" + digilibSizeParams*/ digilibURL, hiddenData, label], true);
```
Comment (roewenstrunk) [inline app/view/window/AnnotationView.js:504]:
```suggestion
                var shape = tplImg.append(div, [id, digilibBaseParams + "dw=600&amp;amp;dh=600" + digilibSizeParams, hiddenData, label], true);
```

File: app/view/window/AnnotationView.js [hunk 10]
```diff
@@ -571,28 +586,31 @@ Ext.define('EdiromOnline.view.window.AnnotationView', {
             var hiddenData = participant['hiddenData'];
             var content = participant['content'];
 
-            label = (siglum == ''?source:siglum) + ": " + label;
+            label = (siglum == ''?source:siglum);//  + ": " + label;
```
Comment (roewenstrunk) [inline app/view/window/AnnotationView.js:589]:
```suggestion
            label = (siglum == ''?source:siglum) + ": " + label;
```

File: app/view/window/Window.js [hunk 11]
```diff
@@ -67,6 +67,9 @@ Ext.define('EdiromOnline.view.window.Window', {
     setWindowConfig: function(config) {
         var me = this;
 
+		//console.log("setWindowConfig");
+		//console.log(config);
+
```
Comment (roewenstrunk) [inline app/view/window/Window.js:72]:
```suggestion
```

File: expath-pkg.xml [hunk 12]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<package xmlns="http://expath.org/ns/pkg" name="Edirom-Online-RWA" abbrev="EdiromOnline-RWA" version="VERSION" spec="1.0">
+    <title>Edirom Online | RWA</title>
```
Comment (roewenstrunk) [inline expath-pkg.xml:3]:
```suggestion
<package xmlns="http://expath.org/ns/pkg" name="Edirom-Online" abbrev="EdiromOnline" version="VERSION" spec="1.0">
    <title>Edirom Online</title>
```

File: index.html [hunk 13]
```diff
@@ -44,15 +44,19 @@
         <script src="app.js"></script>
     <!-- </x-compile> -->   
 
+    <!-- iFrame history -->
+    <script type="text/javascript" src="history.js"/>
+
```
Comment (roewenstrunk) [inline index.html:49]:
```suggestion
```

File: index.html [hunk 14]
```diff
@@ -44,15 +44,19 @@
...
     <!-- **CSS** -->
     <link rel="stylesheet" type="text/css" href="resources/css/tipped/tipped.css" />
     <link rel="stylesheet" type="text/css" href="resources/css/todo.css"/>
+    <link rel="stylesheet" type="text/css" href="resources/css/EdiromOnlineRWA.css"/>
```
Comment (roewenstrunk) [inline index.html:53]:
```suggestion
```

File: packages/eoTheme/sass/var/button/Button.scss [hunk 15]
```diff
@@ -222,16 +223,18 @@ $edirom-taskbarWindowButton-background-color: #cccccc;
 	font-size: 13px;
 	top: 16px;
 	left: 0;
-	content:'Digital';
+	content:'Band II/8';
```
Comment (roewenstrunk) [inline packages/eoTheme/sass/var/button/Button.scss:226]:
```suggestion
	content:'';
```

**Review by roewenstrunk — CHANGES_REQUESTED** [review]
Please delete
* add/data/xslt/meiP2html_rwaOnline.xsl
* add/history.js
* resources/css/EdiromOnlineRWA.css

File: add/data/xql/getAnnotationPreviews.xql [hunk 16]
```diff
@@ -335,6 +349,51 @@ declare function local:toJSON($type as xs:string, $label as xs:string, $mdiv as
...
+                '","part":"',$part,
+                '","page":"',$page,
+                '","source":"',$source,
+                '","siglum":"',$siglum,
+                '","digilibURL":"', $singleURL,
+                '","hiddenData":"',$hiddenData,
+                '","content":"',$content,
+                '","linkUri":"',$linkUri,
+                '"}'
+            )
```
Comment (roewenstrunk) [inline add/data/xql/getAnnotationPreviews.xql:396]:
```suggestion
```

**Review by roewenstrunk — CHANGES_REQUESTED** [review]

**Review by nikobeer — APPROVED** [review]

**Pull request merged**